src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -0,0 +1,259 @@
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>An <see cref="HttpClient"/> instance with the configured handlers.</returns>
+        private static HttpClient Create(HttpMessageHandler innerHandler, params DelegatingHandler[] handlers)
+        {
+            HttpMessageHandler pipeline = CreatePipeline(innerHandler, handlers);
+            HttpClient client = new HttpClient(pipeline);
+            client.DefaultRequestHeaders.Add(SdkVersionHeaderName, SdkVersionHeaderValue);
+            return client;
+        }
+
+
+        /// <summary>
+        /// Creates an instance of an <see cref="HttpMessageHandler"/> using the <see cref="DelegatingHandler"/> instances
+        /// provided by <paramref name="handlers"/>. The resulting pipeline can be used to manually create <see cref="HttpClient"/>
+        /// or <see cref="HttpMessageInvoker"/> instances with customized message handlers.
+        /// </summary>
+        /// <param name="innerHandler">The inner handler represents the destination of the HTTP message channel.</param>
+        /// <param name="handlers">An ordered list of <see cref="DelegatingHandler"/> instances to be invoked as part 
+        /// of sending an <see cref="HttpRequestMessage"/> and receiving an <see cref="HttpResponseMessage"/>.
+        /// The handlers are invoked in a top-down fashion. That is, the first entry is invoked first for 
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>The HTTP message channel.</returns>
+        public static HttpMessageHandler CreatePipeline(HttpMessageHandler innerHandler, IEnumerable<DelegatingHandler> handlers)
+        {
+            if (innerHandler == null)
+            {
+                innerHandler = new HttpClientHandler();
+            }
+
+            if (handlers == null)
+            {
+                return innerHandler;
+            }
+
+            HttpMessageHandler pipeline = innerHandler;
+            IEnumerable<DelegatingHandler> reversedHandlers = handlers.Reverse();
+            foreach (DelegatingHandler handler in reversedHandlers)
+            {
+                if (handler == null)
+                {
+                    throw new ServiceException(
+                    new Error
+                    {
+                        Code = ErrorConstants.Codes.DelegatingHandlerArray,
+                        Message = ErrorConstants.Messages.DelegatingHandlerArrayContainsNullItem,
+                    });
+                }
+
+                if (handler.InnerHandler != null)
+                {
+                    throw new ServiceException(
+                   new Error
+                   {
+                       Code = ErrorConstants.Codes.DelegatingHandlerArrayInnerHandler,
+                       Message = ErrorConstants.Messages.DelegatingHandlerArrayHasNullInnerHandler,
```

Comment:
Are you checking whether the DelegatingHandler already has an innerHandler? Is it an issue if one exists since it is  being changed on line 220?

If so, I think DelegatingHandlerArrayHasNullInnerHandler should be changed to DelegatingHandlerArrayHasUnexpectedInnerHandler and the message should be changed accordingly.